git-chain: init at 0-unstable-2024-08-09

bcyran opened the pull request:
## Description of changes

Adding [`git-chain`](https://github.com/dashed/git-chain) - a tool for working with a chain of dependent git branches. Simplifies some tedious tasks like rebasing many branches after pulling changes from the main branch.

FYI @dashed 

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

bcyran committed:
git-chain: init at 0-unstable-2024-09-09

bcyran committed:
maintainers: add bcyran

Bot-wxt1221 commented on pkgs/by-name/gi/git-chain/package.nix:
```diff
@@ -0,0 +1,36 @@
+
+  src = fetchFromGitHub {
+    owner = "dashed";
+    repo = pname;
```
```suggestion
    repo = "git-chain";
```

bcyran replied:
Done!

Bot-wxt1221 commented on pkgs/by-name/gi/git-chain/package.nix:
```diff
@@ -0,0 +1,36 @@
+  nativeCheckInputs = [ git ];
+
+  meta = with lib; {
+    description = "Tool for rebasing a chain of local git branches";
```
miss changelog here

bcyran replied:
This project doesn't provide any changelog or any kind of releases, not even tags. Not sure what I could link as a changelog.

Bot-wxt1221 replied:
That's ok.

bcyran commented:
@Bot-wxt1221 Do I need to do anything after addressing the comments, like requesting review? And what's up with ofborg's internal error?

Not trying to rush you or anything, I just don't know the usual workflow here.

Bot-wxt1221 commented:
@bcyran You can pin someone who review this PR. Try to rebase the commit to the newest and push again to check if ofborg will work. If not, pin the maintainers of ofborg.

Bot-wxt1221 reviewed (approved):
Consider add update script. But not neccessary.

bcyran commented:
@Bot-wxt1221 Thanks for help and the review!

Added an update script. Testing it actually revealed that I initially made a mistake in the version so it was a good call. Would appreciate you taking a final look.

Bot-wxt1221 commented:
We need an committer anyway.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4600

bcyran commented:
@Bot-wxt1221 @Aleksanaa Hi guys! Sorry to bother you but is it normal for PRs to get stuck for so long? Is there anything I can do to push this forward? Anything I can do to avoid this in the future?

uncenter commented:
> is it normal for PRs to get stuck for so long? Is there anything I can do to push this forward?

Sometimes unfortunately. With such a high volume of PRs, some get missed. It helps to be really active in promoting/getting reviewers to approve and bump your PR in the list. As you can see by https://github.com/NixOS/nixpkgs/pull/342850#issuecomment-2366247982, there are some forum threads and also Discord channels to request reviews.

uncenter commented on pkgs/by-name/gi/git-chain/package.nix:
```diff
@@ -0,0 +1,39 @@
+  openssl,
+  git,
+  nix-update-script,
+}:
```
```suggestion
  stdenv,
  darwin,
}:
```

uncenter commented on pkgs/by-name/gi/git-chain/package.nix:
```diff
@@ -0,0 +1,39 @@
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ];
```
```suggestion
  buildInputs =
    [ openssl ]
    ++ lib.optionals stdenv.hostPlatform.isDarwin (
      with darwin.apple_sdk.frameworks;
      [
        Security
      ]
    );
```

bcyran commented:
Thanks for help @uncenter!

I applied and pushed the requested changes around 12 hours ago but for some reason GitHub is stuck at "Processing updates". I tried rebasing and pushing a few more times but to no avail. I also just noticed that apparently there are still 2 pending CI jobs queued for 20 days now? As a person used to GitLab I'm quite confused by all of this. Do you have any advice how to proceed?

uncenter commented:
Hmm are you getting an error locally when trying to push? Can you explain more about your issue? Is it still occuring?

bcyran commented:
@uncenter The problem was that the pushed changes were visible on my branch but not in the PR. I rebased and pushed one more time right now and the problem is gone so I guess it was some kind of GH lag.

Anyway, your suggestions are applied!
